### PR TITLE
[ATO-711] Port handle deflated webhook request to 3.2.x

### PR DESCRIPTION
**Pull request by radovanZRasa**
**Proposed changes**:
Add capability to handle deflated body of a HTTP request.

**Status (please check what you already did)**:
- [ ] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation in the [rasaHQ/rasa](https://github.com/rasaHQ/rasa)
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa-sdk/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa-sdk#code-style) for instructions)

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -80,9 +69,6 @@ jobs:
       - name: Check Types 📚
         run: make types
 
-      - name: Test Docs 🕸
-        run: make test-docs
-
```
Comment (m-vdb) [inline .github/workflows/continuous-integration.yml:85]: can you remove [these 2 make targets](https://github.com/RasaHQ/rasa-sdk/blob/main/Makefile#L58-L67) too? I think they're unused now

**Review by m-vdb — COMMENTED** [review]
I'll let Tom or Anca review the actual code ;)

File: rasa_sdk/endpoint.py [hunk 2]
```diff
@@ -90,7 +92,13 @@ async def health(_) -> HTTPResponse:
     @app.post("/webhook")
     async def webhook(request: Request) -> HTTPResponse:
```
Comment (m-vdb) [inline rasa_sdk/endpoint.py:93]: [the test is missing](https://github.com/RasaHQ/rasa-sdk/pull/925/files#diff-adf6af4744d4ee23f8b03b6fddcd7ca0ae330cd2ba8e3dc39c834ee76dbe47de)

**Review by m-vdb — CHANGES_REQUESTED** [review]

**Review by m-vdb — APPROVED** [review]

**Review by tmbo — APPROVED** [review]
lgtm

**Pull request merged**